xio/decorator.go

```diff
@@ -55,11 +55,16 @@ func RateLimit(limit int) WriterDecorator {
 
 // SizeLimit decorator is used to add a size limit (number of bytes) for io.Writer.
 // If written bytes length exceeds the passed limit it will return an error.
-func SizeLimit(size int) WriterDecorator {
+func SizeLimit(size int, allow bool) WriterDecorator {
 	return func(writer io.Writer) io.Writer {
 		return WriterFunc(func(p []byte) (int, error) {
 			if len(p) > size {
-				return 0, ErrSizeLimitExceeded
+				if allow {
+					_, _ = writer.Write(p)
```

Comment:
Catch any errors and log them like on line 187. Also, if the writer wrote anything, use that for return even with ErrSizeLimitExceeded